Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM docker.io/library/debian:bullseye-slim as base
+ENV SHELL /bin/bash
+
+WORKDIR /opt
+RUN apt-get -y update --fix-missing && \
+    apt-get -y upgrade && \
+    apt-get install -y --no-install-recommends \
+    libssl-dev make cmake graphviz \
+    git pkg-config curl time rhash ca-certificates jq \
+    python3 python3-pip lsof ruby ruby-bundler git-restore-mtime xz-utils zstd unzip gnupg protobuf-compiler \
+    wget net-tools iptables iproute2 iputils-ping ed zlib1g-dev wabt && \
+    curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
+    apt-get install -y nodejs
+
+# foundry
+RUN curl -L https://foundry.paradigm.xyz/ | bash && \
+    ${HOME}/.foundry/bin/foundryup
+# go
+RUN wget https://dl.google.com/go/go1.20.4.linux-amd64.tar.gz && \
+    # -C to move to given directory
+    tar -C /usr/local/ -xzf go1.20.4.linux-amd64.tar.gz
+# pnpm
+RUN npm install pnpm --global
+
+FROM base AS builder
+COPY . /app
+WORKDIR /app
+
+ENV PATH="${PATH}:/usr/local/go/bin"
+ENV PATH="${PATH}:/root/.foundry/bin"
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
+RUN pnpm run -r build
+
+FROM builder AS store-indexer
+WORKDIR /app/packages/store-indexer
```

Comment:
since this dockerfile is specific to store-indexer, should we put this in `packages/store-indexer/bin`? or does that make the rest of the infra complicated?

Reply:
we are having one top level docker file that builds the entire monorepo, then takes specific artifacts and put them in different images. This dockerfile actually describes two images right now: a global `builder`, and a specific `store-indexer`. Over time we will add more of them, one for each TS service that requires a Docker image.
having one per package is actually worse, given we would need to build the entire monorepo for each of them (instead of looping through all targets -- currently only `store-indexer`, and building + tagging them individually which will leverage the cache of the build stage).